src/server/scripts/Kalimdor/TempleOfAhnQiraj/temple_of_ahnqiraj.cpp

```diff
@@ -183,8 +189,119 @@ class spell_aggro_drones : public SpellScript
     }
 };
 
+struct npc_obsidian_nullifier : public ScriptedAI
+{
+    npc_obsidian_nullifier(Creature* creature) : ScriptedAI(creature)
+    {
+    }
+
+    void Reset() override
+    {
+        _scheduler.CancelAll();
+        me->SetPower(POWER_MANA, 0);
+        _targets.clear();
+    }
+
+    void EnterCombat(Unit* /*who*/) override
+    {
+        _scheduler.Schedule(6s, [this](TaskContext context)
+        {
+            if (_targets.empty())
+            {
+                Map::PlayerList const& players = me->GetMap()->GetPlayers();
+                for (Map::PlayerList::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+                {
+                    if (Player* player = itr->GetSource())
+                    {
+                        if (player->IsAlive() && !player->IsGameMaster() && !player->IsSpectator() && player->GetPower(POWER_MANA) > 0)
+                        {
+                            _targets.push_back(player);
+                        }
+                    }
+                }
+
+                Acore::Containers::RandomResize(_targets, 11);
+            }
+
+            for (Unit* target : _targets)
+            {
+                DoCast(target, SPELL_DRAIN_MANA, true);
+            }
+
+            if (me->GetPowerPct(POWER_MANA) >= 100.f)
+            {
+                DoCastAOE(SPELL_NULLIFY, false);
+            }
+
+            context.Repeat(6s);
+        })
+        .Schedule(6000ms, 8400ms, [this](TaskContext context)
+        {
+            DoCastVictim(SPELL_CLEAVE, true);
+            context.Repeat(6000ms, 8400ms);
+        });
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+        {
+            return;
+        }
+
+        _scheduler.Update(diff,
+            std::bind(&ScriptedAI::DoMeleeAttackIfReady, this));
+    }
+
+private:
+    TaskScheduler _scheduler;
+    std::list<Player*> _targets;
+};
+
+class spell_drain_mana : public SpellScript
```

Comment:
https://github.com/azerothcore/azerothcore-wotlk/blob/e623e9ef49bdcc986cc32683ada3911820f858a5/src/server/scripts/Kalimdor/RuinsOfAhnQiraj/ruins_of_ahnqiraj.cpp#L147 Their functions are the same, but they will conflict if they exist at the same time